File: backend/app/schemas/seller_knowledge.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.seller_knowledge import SellerKnowledgeStatus


class SellerDocumentOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    filename: str
    file_type: str
    file_size: int
    created_at: datetime


class UpdateSellerProfileRequest(BaseModel):
    company_name: str | None = None
    company_website: str | None = None
    product_website: str | None = None
    additional_notes: str | None = None


class SellerKnowledgeOut(BaseModel):
    id: uuid.UUID
    company_name: str | None
    company_website: str | None
    product_website: str | None
    additional_notes: str | None
    status: SellerKnowledgeStatus
    confidence: float | None
    model_name: str | None
    prompt_version: str | None
    generated_at: datetime | None
    failure_reason: str | None
    sources_processed: int | None
    knowledge: dict | None
    documents: list[SellerDocumentOut]

File: frontend/lib/seller-knowledge.ts
export type SellerDocument = {
  id: string;
  filename: string;
  file_type: string;
  file_size: number;
  created_at: string;
};

export type SellerKnowledgeData = {
  company_summary: string;
  product_summary: string;
  ideal_customer_profile: string[];
  primary_industries: string[];
  business_problems_solved: string[];
  core_capabilities: string[];
  differentiators: string[];
  competitive_advantages: string[];
  pricing_position: string;
  buyer_personas: string[];
  cost_savings: string[];
  time_savings: string[];
  automation_opportunities: string[];
  customer_outcomes: string[];
  proof_points: string[];
  discovery_questions: string[];
  common_objections: string[];
  why_customers_switch: string[];
  recommended_pitch: string;
  confidence: number;
};

export type SellerKnowledgeEnvelope = {
  data: SellerKnowledgeData;
  meta: { model_name: string; prompt_version: string; generated_at: string };
};

export type SellerKnowledgeStatusValue = "NOT_GENERATED" | "GENERATING" | "COMPLETED" | "FAILED";

export type SellerKnowledgeProfile = {
  id: string;
  company_name: string | null;
  company_website: string | null;
  product_website: string | null;
  additional_notes: string | null;
  status: SellerKnowledgeStatusValue;
  confidence: number | null;
  model_name: string | null;
  prompt_version: string | null;
  generated_at: string | null;
  failure_reason: string | null;
  sources_processed: number | null;
  knowledge: SellerKnowledgeEnvelope | null;
  documents: SellerDocument[];
};
